src/pages/ScholarshipForm.jsx

```diff
@@ -1,115 +1,134 @@
-import React, { useState } from 'react';
+import React from 'react';
+import { useFormik, getIn } from 'formik';
 import { Container, Button, TextField } from '@material-ui/core';
 import { makeStyles } from '@material-ui/core/styles';
 import ScholarshipAmountField from '../components/ScholarshipAmountField';
 import DatePicker from '../components/DatePicker';
-import { invalidAmountFields } from '../validation/amountValidation';
+import { validationSchema } from '../validation/ValidationSchema';
 
-const useStyles = makeStyles({
+const useStyles = makeStyles((theme) => ({
   containerStyle: {
     maxWidth: 400,
   },
-});
-
-// regular expression used for valid website urls
-const websiteReg = /^(http(s)?:\/\/(www\.)?)?[\w-]+(\.[\w-]+)*\.[a-z]{2,}((\/|#|\?)(\S)*)?$/;
+  fieldStye: {
+    paddingBottom: theme.spacing(1),
+  },
+}));
 
 function ScholarshipForm() {
   const classes = useStyles();
-  const [values, setValues] = useState({
-    name: '',
-    deadline: null,
-    description: '',
-    amountType: null,
-    minAmount: 0,
-    maxAmount: 0,
-    website: '',
-    scholarshipType: '',
-  });
-
-  const [errors, setErrors] = useState({});
 
-  function validateForm() {
-    /* eslint-disable no-restricted-globals */
-    setErrors({
-      name: !values.name,
-      deadline: !values.deadline || isNaN(values.deadline),
-      description: !values.description,
-      website: !websiteReg.test(values.website),
-      amountHelperText: invalidAmountFields(
-        values.amountType,
-        values.minAmount,
-        values.maxAmount
-      ),
-    });
-  }
+  const formik = useFormik({
+    initialValues: {
+      name: '',
+      deadline: null,
+      description: '',
+      amount: {
+        type: null,
+        minAmount: 0,
+        maxAmount: 0,
+      },
+      website: '',
+    },
+    validationSchema,
+    onSubmit: (values, { setSubmitting, resetForm }) => {
+      setSubmitting(true);
+      alert(JSON.stringify(values, null, 2));
+      setSubmitting(false);
+      resetForm();
+    },
+  });
 
   function updateAmount(minAmount, maxAmount) {
-    setValues({ ...values, minAmount, maxAmount });
+    formik.setFieldValue('amount.minAmount', minAmount, false);
+    formik.setFieldValue('amount.maxAmount', maxAmount, false);
   }
 
   return (
     <Container maxWidth="md">
-      <form className={classes.containerStyle}>
+      <form className={classes.containerStyle} onSubmit={formik.handleSubmit}>
         <h1>Submit a Scholarship</h1>
         <div>
           <TextField
+            className={classes.fieldStye}
             id="name"
-            label="Scholarship Name"
-            error={errors.name}
-            helperText={errors.name ? 'Please enter a name' : ' '}
-            required
-            value={values.name}
-            onChange={(e) => setValues({ ...values, name: e.target.value })}
+            label="Scholarship Name *"
```

Comment:
Is the `*` necessary if you still pass in the `required` field?

Reply:
I took out the required field because It was triggering the built in validation on submit, instead of using Formik.

Reply:
(I'm ok either way)

Reply:
Ah alright